web/src/stream_list_sort.ts

```diff
@@ -1,16 +1,19 @@
+import assert from "minimalistic-assert";
+
 import * as settings_config from "./settings_config";
 import * as stream_data from "./stream_data";
 import * as stream_topic_history from "./stream_topic_history";
 import * as sub_store from "./sub_store";
+import type {StreamSubscription} from "./sub_store";
 import {user_settings} from "./user_settings";
 import * as util from "./util";
 
-let previous_pinned;
-let previous_normal;
-let previous_dormant;
-let previous_muted_active;
-let previous_muted_pinned;
-let all_streams = [];
+let previous_pinned: number[] | undefined;
+let previous_normal: number[] = [];
```

Comment:
I made these not undefined anymore, since there wasn't an undefined check where they were being used, and it seems like that works fine.

In fact, maybe it makes sense to also make `previous_pinned` never undefined and have some different flag than `previous_pinned === undefined`, since it seems like that's just checking if all these arrays have been assigned value yet.

Reply:
Yeah I think a prep commit introducing a `first_render` boolean that starts as `false` and replacing the `previous_pinned !== undefined` check with a `!first_render` check would work fine.

Reply:
Did that (though with a slightly different name to be more, and I think you'd want a `first_render` not `!first_render`)